internal/datastore/mysql/datastore_test.go

```diff
@@ -711,6 +711,23 @@ func TestMySQLMigrationsWithPrefix(t *testing.T) {
 	req.NoError(rows.Err())
 }
 
+func TestMySQLWithAWSIAMCredentialsProvider(t *testing.T) {
```

Comment:
here's a take at adding a test for this

alternatively, we could expect authentication to fail if we try connecting to the actual test datastore